integration-tests/tests/src/node/ssl.ts

```diff
@@ -0,0 +1,253 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+const { X509Certificate } = await import("node:crypto");
+import {
+  ConfigurationWithSync,
+  Credentials,
+  ErrorCallback,
+  ProgressRealmPromise,
+  Realm,
+  SessionStopPolicy,
+  SSLConfiguration,
+  SSLVerifyObject,
+  User,
+} from "realm";
+
+import { closeRealm } from "../utils/close-realm";
+import { createPromiseHandle } from "../utils/promise-handle";
+import { importAppBefore } from "../hooks";
+import { PersonSchema } from "../schemas/person-and-dog-with-object-ids";
+
+// IMPORTANT:
+// * Can only run on non-Apple machines, otherwise tests will await forever.
+
+describe.skipIf(environment.missingServer, "SSL Configuration", function () {
```

Comment:
Perhaps we can also figure out a way to `skipIf` the users machine is a mac?

Reply:
I've tried locally on my Mac now and for MacOS we can import `node:os` and use:
```js
describe.skipIf(platform() === "darwin" || environment.missingServer, ..)
```

Reply:
That's great!  Let's do that 🙂

Reply:
✅